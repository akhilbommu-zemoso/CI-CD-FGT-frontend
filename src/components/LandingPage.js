import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Grid from '@material-ui/core/Grid';
import BudgetCard from './AddBudgetCard'
import InsightCard from './InsightCard'
import TransactionsCard from './TransactionsCard'
const useStyles = makeStyles((theme) => ({
    root: {
        flexGrow: 1,
    },
    paper: {
        padding: theme.spacing(2),
        textAlign: 'center',
        color: theme.palette.text.secondary,
    },
    Top: {
        marginTop: '200px',
    },
    title: {
        marginBottom: '5px',
        fontSize: '20px',
        fontWeight: 500,
        color: 'black',
        marginLeft: '10pt'
    },
    top: {
        marginTop:'25px',
    }
}));
export default function CenteredGrid() {
    const classes = useStyles();
    return (
        <div className={classes.root}>
            <Grid container spacing={12}>
                <Grid item xs={9} className={classes.left}>
                    <div className={classes.title}>
                        Budget
                    </div>
                    <BudgetCard title="Create your budget" content=" Take control on your expenses by creating a budget on your spendings" />
                </Grid>
                <Grid item xs={3} className={classes.left}>
                    <div className={classes.title}>
                    </div>
                    <InsightCard title="Insights" content="No insights are here"/>
                </Grid>
                <Grid item xs={9} className={classes.left}>
                    <div className={classes.title}>
                    </div>
                </Grid>
                <Grid item xs={3} className={classes.left}>
                    <div className={classes.title}>
                    </div>
                    <div className={classes.top}>
                        <TransactionsCard title="Transactions" content="Looks like you didnâ€™t create any transactions to show the data. Please, here to keep in track"/>
                    </div>
                </Grid>
            </Grid>
        </div>
    );
}





    
