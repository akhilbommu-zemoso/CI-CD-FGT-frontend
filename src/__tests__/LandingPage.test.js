import React from 'react'
import { render, fireEvent, screen,getAllByText } from '@testing-library/react'
import LandingPage from '../components/LandingPage'
import { BrowserRouter as Router } from 'react-router-dom'

it('should take a snapshot', () => {
    const { asFragment, getByTestId } = render(<Router><LandingPage /></Router>)
    expect(asFragment(<Router><LandingPage /></Router>)).toMatchSnapshot()

});

it("checks for Budget card", () => {
    const { getByText, getByTestId } = render(<Router><LandingPage /></Router>)
    expect(screen.getByText("Create your budget")).toBeInTheDocument
})

it("checks for transactions card", () => { 
    const { getByText ,getByTestId } = render(<Router><LandingPage /></Router>)
    expect(screen.getByText("Trasactions")).toBeInTheDocument
})

it("checks for Insights card", () => {
    const { getByText, getByTestId } = render(<Router><LandingPage /></Router>)
    expect(screen.getByText("Insights")).toBeInTheDocument
})