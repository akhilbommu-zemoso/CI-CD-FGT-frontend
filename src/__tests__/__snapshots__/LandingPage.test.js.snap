// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-root-1"
  >
    <div
      class="MuiGrid-root MuiGrid-container"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-9"
      >
        <div
          class="makeStyles-title-4"
        >
          Budget
        </div>
        <div
          class="MuiPaper-root MuiCard-root makeStyles-cardStyle-109 MuiPaper-elevation1 MuiPaper-rounded"
        >
          <div
            class="MuiCardContent-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body1"
            >
              Create your budget
            </p>
            <p
              class="MuiTypography-root makeStyles-content-110 MuiTypography-body1"
            >
              Take control on your expenses by creating a budget on your spendings
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1"
            />
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
      >
        <div
          class="makeStyles-title-4"
        />
        <div
          class="MuiPaper-root MuiCard-root makeStyles-card-171 MuiPaper-elevation1 MuiPaper-rounded"
        >
          <div
            class="MuiCardContent-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body1"
            />
            <div>
              <div>
                <p
                  class="MuiTypography-root makeStyles-header-172 MuiTypography-body1"
                >
                  Insights
                </p>
                <div />
              </div>
              <hr
                class="MuiDivider-root"
                style="margin-top: 10px;"
              />
            </div>
            <p />
            <p
              class="MuiTypography-root makeStyles-content-174 MuiTypography-body1"
            >
              <img
                alt="Logo"
                height="80px"
                src="insight.png"
              />
            </p>
            <p
              class="MuiTypography-root makeStyles-footer-173 MuiTypography-body1"
            >
              No insights are here
            </p>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-9"
      >
        <div
          class="makeStyles-title-4"
        />
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
      >
        <div
          class="makeStyles-title-4"
        />
        <div
          class="makeStyles-top-5"
        >
          <div
            class="MuiPaper-root MuiCard-root makeStyles-card-182 MuiPaper-elevation1 MuiPaper-rounded"
          >
            <div
              class="MuiCardContent-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body1"
              />
              <div>
                <div
                  class="makeStyles-header-184"
                >
                  <p
                    class="MuiTypography-root makeStyles-title-183 MuiTypography-body1"
                  >
                    Trasactions
                  </p>
                  <div>
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root makeStyles-iconStyle-186"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                      />
                    </svg>
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root makeStyles-iconStyle-186"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M20 3h-1V1h-2v2H7V1H5v2H4c-1.1 0-2 .9-2 2v16c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 18H4V8h16v13z"
                      />
                    </svg>
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root makeStyles-iconStyle-186"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M3 18h6v-2H3v2zM3 6v2h18V6H3zm0 7h12v-2H3v2z"
                      />
                    </svg>
                  </div>
                </div>
                <hr
                  class="MuiDivider-root"
                  style="margin-top: 10px;"
                />
              </div>
              <p />
              <p
                class="MuiTypography-root makeStyles-content-185 MuiTypography-body1"
              >
                Looks like you didnâ€™t create any transactions to show the data. Please, here to keep in track
              </p>
              <p
                class="MuiTypography-root MuiTypography-body1"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
